src/System.Text.Json/src/System/Text/Json/Serialization/PooledBufferWriter.cs

```diff
@@ -98,18 +101,30 @@ public void Advance(int count)
             _index += count;
         }
 
-        public Memory<T> GetMemory(int sizeHint = 0)
+        public Memory<byte> GetMemory(int sizeHint = 0)
         {
             CheckAndResizeBuffer(sizeHint);
             return _rentedBuffer.AsMemory(_index);
         }
 
-        public Span<T> GetSpan(int sizeHint = 0)
+        public Span<byte> GetSpan(int sizeHint = 0)
         {
             CheckAndResizeBuffer(sizeHint);
             return _rentedBuffer.AsSpan(_index);
         }
 
+#if BUILDING_INBOX_LIBRARY
+        internal ValueTask WriteToAsync(Stream destination, CancellationToken cancellationToken)
+        {
+            return destination.WriteAsync(WrittenMemory, cancellationToken);
```

Comment:
nit: Make these methods actually async (and await).

Reply:
> nit: Make these methods actually async (and await).

Why?

Reply:
If the only thing the method does is returning an awaitable task, there is no need to await it twice.

```cs
Task MethodAsync() => stream.WriteAsync(); // the method just returns the task, no extra code generated
await MethodAsync(); // single async state machine created
```


```cs
async Task MethodAsync() => await stream.WriteAsync(); // first async state machine created
await MethodAsync(); // second async state machine created
```

@stephentoub do we have some docs with best practices for async?

Reply:
Looks like we use method names suffixed with `Async` even when they aren't actually async (i.e. use the C# keyword) and just return a Task/ValueTask, which does make sense. I retract my nit.